types/artifacts.d.ts

```diff
@@ -179,6 +181,21 @@ declare global {
         params: {name: string; value: string}[];
       }
 
+      export interface IFrameElement {
+        /** The `id` attribute of the IFrame. */
+        id: string,
+        /** The `src` attribute of the IFrame. */
+        src: string,
+        /** The IFrame's ClientRect. @see https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect */
+        clientRect: ClientRect,
```

Comment:
we've had issues with putting some of the real DOM types in artifacts because of the extra constraints sometimes put on them, e.g. for `ClientRect`, `width` and `height` are marked `readonly`, which can pose an annoyance e.g. in the type-checked tests, scripts that overwrite properties on artifacts, etc.

Since it's not really a `ClientRect` anyways (we pass it through JSON via the protocol), can you switch this to a manually specified object like `ImageElement.clientRect` does (though with `width` and `height`, since I vaguely recall that you do use those)?